web/src/emoji_picker.js

```diff
@@ -601,6 +611,34 @@ function register_popover_events($popover) {
 
     $(".emoji-popover-filter").on("input", filter_emojis);
     $(".emoji-popover-filter").on("keydown", process_enter_while_filtering);
+    $(".emoji-popover-filter").on("keydown", () => {
+        const $search_input = $("#emoji-search-query");
+        const $clear_button = $("#emoji_search_clear");
+        $(document).on("input", "#emoji-search-query", () => {
```

Comment:
Why are these event listeners nested? If I remember right, event listeners do apply to hidden elements since they do in fact exist in the DOM. If they didn't we could use event delegation to make it work, instead of these nested event listeners attached on the `document`.